airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -150,6 +156,22 @@ public class PostgresSource extends AbstractJdbcSource<PostgresType> implements
   public static final String SSL_MODE_DISABLE = "disable";
   public static final String SSL_MODE_REQUIRE = "require";
 
+  public static final Map<PGProperty, String> JDBC_CONNECTION_PARAMS = ImmutableMap.of(
+      // Initialize parameters with prepareThreshold=0 to mitigate pgbouncer errors
+      // https://github.com/airbytehq/airbyte/issues/24796
+      PREPARE_THRESHOLD, "0",
+      DEFAULT_ROW_FETCH_SIZE, "1",
+      ADAPTIVE_FETCH, "true",
+      MAX_RESULT_BUFFER, "10percent",
```

Comment:
this sets a limit of 10% of RAM / up to 1000 rows pet fetch?
Wouldn't that slow us down? we typically fetch rows numbered in 100Ks if not millions.

Reply:
I wondered about that. Mysql fetches 1 row at a time, and SQL Server fetches 10 rows. Perhaps we can run a performance test. Let's discuss on Monday, in the meantime we should not merge this on a Friday.

Reply:
> Mysql fetches 1 row at a time, and SQL Server fetches 10 rows 
I don't think that's the case. we start by taking a sample of 10. But AFAIK current fetching limit is attempting to fill up to 50% of RAM

Reply:
do we have any performance benchmark we can use as a quick regression test?

Reply:
Curious here : why the significant departure from our default fetch settings? 
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/streaming/FetchSizeConstants.kt

Reply:
The comments I committed should explain this.

It's very possible that this adaptiveFetch feature just plain sucks and is inferior to our own. Lets see if the postgres performance takes a hit and worst case we revert this PR.